# HG changeset patch
# Parent 01b10d7cdf99c6de50a044438c7f292b405d371a
Create usefulness in rc.subr etc/rc.conf.d/*.conf namespace.

diff -r 01b10d7cdf99 etc/rc.subr
--- a/etc/rc.subr
+++ b/etc/rc.subr
@@ -1055,9 +1055,19 @@
 		fi
 		_rc_conf_loaded=true
 	fi
-	if [ -f /etc/rc.conf.d/"$_name" ]; then
-		debug "Sourcing /etc/rc.conf.d/${_name}"
-		. /etc/rc.conf.d/"$_name"
+	if [ -d /etc/rc.conf.d ]; then
+		if [ -f /etc/rc.conf.d/"$_name" ]; then
+			debug "Sourcing /etc/rc.conf.d/${_name}"
+			. /etc/rc.conf.d/"$_name"
+		fi
+
+		local _modular_conf
+		for _modular_conf in /etc/rc.conf.d/*.conf; do
+			if [ -f "$_modular_conf" ]; then
+				debug "Sourcing $_modular_conf"
+				. "$_modular_conf"
+			fi
+		done
 	fi
 
 	# Old variable names support
